### ISSUE TITLE
📞 Split 'Leave call` button for moderators

### ISSUE THREAD

**Issue by Antreesy** [nextcloud/spreed#10343]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are interested into the same feature.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments.

---

**Is your feature request related to a problem? Please describe.**
Moderators don't have an opportunity to immediately leave call, because they have to pick between "end call for me" and "end call for everyone"

**Describe the solution you'd like**
Implement split button pattern:
![image](https://github.com/nextcloud/spreed/assets/93392545/c74eb38c-a20f-4b21-92c3-8e48ac8a15e1)

**Describe alternatives you've considered**
Upstream behavior for `<NcActions />`?

**Comment by nickvergessen** [comment]
cc @marcoambrosini you remember where we discussed this?

**Comment by marcoambrosini** [comment]
@nickvergessen https://github.com/nextcloud/spreed/issues/5395#issuecomment-872824782

**Comment by nickvergessen** [comment]
So ref https://github.com/nextcloud/spreed/pull/5981#issuecomment-927581837 it was a design decision.

so cc @nextcloud/designers any plans to overhaul this decision? Seems every frontend dev feels like it could be useful over and over again :D

---

> Btw the first link is a comment hidden by you, tired of me advocating for the damn split button 😅

Yeah because it was decided otherwise by design and I didn't want wrong implementations because a screenshot showed it :P

**Comment by Antreesy** [comment]
Variation of NcActions with styles changes will be applicable (not sure we need it upstream, but possible to discuss):

```vue
<NcActions :force-name="true" :inline="1">
	<template #icon>
				<ChevronDown :size="20" />
			</template>
		<NcActionButton @click="endCall">
			End call
		</NcActionButton>
		<NcActionButton @click="endCallForEveryone">
			End call for everyone
		</NcActionButton>
	</NcActions>
```

**Comment by jancborchardt** [comment]
Neither Material Design nor Apple Human Interface Guidelines define split buttons, so they are not really in wide use. This is exactly my worry: If we create a component that is not used by the majority of interfaces just so we (who live and breathe the software) have 1 click less, it will result in confusion for regular people who use the software once a week or month.

- Material Design buttons: https://m3.material.io/components/all-buttons
- Closest in the Apple HIG are pull-down buttons, which do exactly what we currently do: https://developer.apple.com/design/human-interface-guidelines/pull-down-buttons

**Comment by marcoambrosini** [comment]
So let's take a step back. I see two issues with the current state. One bigger and one smaller:

1 (bigger): every time I want to leave a call in a conversation in which I'm a moderator I come dangerously close to ending the call for everyone
2 (smaller): It takes 2 clicks to leave a call

3 possible solutions for this that do not require the split button: 
1) Only show a `leave call` and add an `end call for everyone` button in the call actions menu 
2) Keep the current button but show a confirm dialog upon clicking end call for everyone 
3) 1 and 2 combined

**Comment by nickvergessen** [comment]
> Since it’s a very destructive action, it should go last in the action menu, probably even with a divider above.

I don't really like this. Most of the calls should be ended using "End for everyone", to avoid everyone having to terminate their connection on their own. There were also even requests to make this the default action.
I agree to not do that, but it should be rather prominent and discoverable.
When you are managing a meeting, you will not skip through all the menus that exists to see if there is a way to improve the meeting experience for everyone. You might even be nervous or stressed instead, so it has to be easily visible/accessible.

**Comment by jancborchardt** [comment]
> When you are managing a meeting, you will not skip through all the menus that exists to see if there is a way to improve the meeting experience for everyone. You might even be nervous or stressed instead, so it has to be easily visible/accessible.

Yeah, good point as well, people might just not look for it at all. Do you have further proposals or requirements you can think of?

**Comment by marcoambrosini** [comment]
Then I think that the do you really want to end meeting for everyone is almost mandatory

**Comment by marcoambrosini** [comment]
🥳
